demos/SDO-OH-EXF-dev-simple/README.md

```diff
@@ -0,0 +1,97 @@
+apt-get -y upgrade
+apt-get -y install jq
+```
+
+### Step Two: Install Open Horizon
+
+> If Open Horizon is already installed and configured properly, skip step two.
+
+* **Optional** - Determine the VM's local IP address and `export HZN_LISTEN_IP=<local-IP-address-here>` if the network connection is bridged.  If the IP address is accessible outside of the VM, then you will be able to connect to any exposed ports from outside the VM.  Without this step, the services will only be available inside the VM.
+* Install the all-in-one Open Horizon instance by following [these instructions](https://github.com/open-horizon/devops/tree/master/mgmt-hub).
+* **NOTE** Save the credentials shown at the end of the installation process.  You will need these later.
+* Set the following environment variables to connect to the Exchange and to allow publishing.  Replace the `<values>` with the actual value indicated.  Also add them to your shell's RC file if you'd like them to persist beyond this session.
+``` shell
+export HZN_ORG_ID=myorg
+export HZN_EXCHANGE_USER_AUTH=admin:<insert value from installation credentials>
+export HZN_DEVICE_TOKEN=<insert value from installation credentials or use random string>
+export HZN_DEVICE_ID=node1
+export HZN_EXCHANGE_URL=http://<local-IP-address-here>:3090/v1/
+export HZN_FSS_CSSURL=http://<local-IP-address-here>:9443/
+```
+
+### Step Three: Configure and publish the definition files for the EXF Service
+
+* Clone the EdgeX Foundry integration repository and navigate to the `configs` folder:
+``` shell
+git clone https://github.com/edgexfoundry-holding/orra.git
+cd orra/demos/SDO-OH-EXF-dev-simple
+```
+* Publish the service and pattern:
+``` shell
+hzn exchange service publish -P -f configuration/service.json
+hzn exchange pattern publish -f configuration/pattern.json
+```
+* Create the temporary and configuration folders, copy the EdgeX Foundry service configuration patterns to their expected location:
+``` shell
+mkdir -p /var/run/edgex/logs
+mkdir -p /var/run/edgex/data
+mkdir -p /var/run/edgex/consul/data
+mkdir -p /var/run/edgex/consul/config
+mkdir -p /root/res
+chmod -R a+rwx /var/run/edgex
+chmod -R a+rwx /root/res
+cp res/* /root/res
+```
+* Publish the deployment policy:
+``` shell
+hzn exchange deployment addpolicy -f configuration/deployment.policy.json myorg/policy-com.github.joewxboy.horizon.edgex_1.0.1
+```
+
+### Step Four: Edit and run the SDO script
+
+* Go back to your starting folder:
+``` shell
+cd ../../..
+```
+* Edit the `test-sdo.sh` script to change the following lines:
+    * Approximately at line 203: `hzn voucher import /var/sdo/voucher.json --policy node.policy.json`
+    * Change the policy file to `orra/demos/SDO-OH-EXF-dev-simple/configuration/node.policy`
+    * Approximately at line 210: `/usr/sdo/bin/owner-boot-device ibm.helloworld`
+    * Change the pattern from "ibm.helloworld" to `com.github.joewxboy.horizon.edgex`
+* Run the test script:
+``` shell
+./test-sdo.sh
+```
```

Comment:
Suggest to add some commentary during run of this script as to what is happening.

Reply:
I agree and it'd be good to let people know not to worry if they see:
```
26-04-2021 12:50:39.072 WARN  [main] org.sdo.pri.device.DeviceApp.sslContext - UNSAFE: no-op TrustManager installed
```
and
```
26-04-2021 12:50:53.800 ERROR [main] o.s.p.DeviceTransferOwnershipClient.call - (POST http://127.0.0.1:8040/mp/113/msg/30) 500 {"timestamp":"2021-04-26T12:50:53.370+00:00","status":500,"error":"Internal Server Error","message":"","path":"/mp/113/msg/30"}
26-04-2021 12:50:53.879 INFO  [main] o.s.p.DeviceTransferOwnershipClient.call - All rendezvous instructions exhausted. Pausing until Apr 26, 2021, 12:52:48 PM
```
It eventually worked out all right after that pause but it left me wondering what had gone wrong while it was on pause. It would actually be good to tell people what to look for for a sign of successful run.

Reply:
Good point.  Added wording to that effect.